### PR TITLE
Subscription models update + db update + new sub view

### PR DESCRIPTION
**Pull request by ahmed93**

### CODE REVIEW

File: Shared/Views/NewSubView.swift [hunk 1]
```diff
@@ -129,71 +138,62 @@ fileprivate struct OptionsCell: View {
     }
 }
 
-
 fileprivate struct CyclePickerCell: View {
+    @EnvironmentObject var partialSheetManager: PartialSheetManager
```
Comment (eldesouky) [inline Shared/Views/NewSubView.swift:152]: Doesn't need to be an EnviromentalObject. should just be a state object

File: Shared/Views/NewSubView.swift [hunk 2]
```diff
@@ -7,76 +7,81 @@
...
+                ZStack {
+                    viewModel.model.color.opacity(0.5).edgesIgnoringSafeArea(.all)
+                    
+                    viewModel.image
+                        .resizable()
+                        .aspectRatio(contentMode: .fit)
+                        .background(Color.clear)
+                        .padding(10)
+                    
+                    Text("Tab To\nChange")
```
Comment (eldesouky) [inline Shared/Views/NewSubView.swift]: Localize

File: Shared/Views/NewSubView.swift [hunk 3]
```diff
@@ -7,76 +7,81 @@
...
-                .keyboardType(.decimalPad)
-            
-            
-//            Text("\(PresistanceController.isPreviewing)")
+                .onTapGesture(perform: {
+                    // open icons and set value imageName
+                    viewModel.imageName = "netflix_logo"
+                })
+                
+                CurrencyTextField("Amount", value: $viewModel.model.amount, alwaysShowFractions: false, numberOfDecimalPlaces: 2, currencySymbol: viewModel.model.currency.symbol)
```
Comment (eldesouky) [inline Shared/Views/NewSubView.swift]: localize

File: Shared/Views/NewSubView.swift [hunk 4]
```diff
@@ -129,71 +138,62 @@ fileprivate struct OptionsCell: View {
     }
 }
 
-
 fileprivate struct CyclePickerCell: View {
+    @EnvironmentObject var partialSheetManager: PartialSheetManager
+
     @State var title: AppLocal.Strings
-//    @Binding var value: String
+    @Binding var value: Cycle
```
Comment (eldesouky) [inline Shared/Views/NewSubView.swift]: Here you should use a protocol CyclePickerContentDisplay protocol, where Cycle and Duration implement this protocol. So same View can easily be used for both input fields.

File: Shared/Models/Subscription+CoreDataClass.swift [hunk 5]
```diff
@@ -52,4 +52,21 @@ extension Subscription {
...
     
+    func model() -> SubscriptionModel {
+        return .init(dbModel: self)
+    }
+    
+    func update(model: SubscriptionModel) {
+        let subType = SubscriptionType.allCases[Int.random(in: 0 ... SubscriptionType.allCases.count - 1)]
+        self.amount = model.amount
+        self.currency = model.currency.locale.identifier
+        self.cycle = "\(model.cycle.number.intValue)-\(model.cycle.period.rawValue)"
```
Comment (eldesouky) [inline Shared/Models/Subscription+CoreDataClass.swift]: are you really just planning to cycle and duration it as strings?
then casting them back to enums ?

File: Shared/Views/NewSubView.swift [hunk 6]
```diff
@@ -7,76 +7,81 @@
...
-                viewModel.image
-                    .resizable()
-                    .aspectRatio(contentMode: .fit)
-                    .background(Color.clear)
-                    .padding()
+        HStack(alignment: .center) {
+            Spacer()
+            VStack(spacing: 5) {
+                ZStack {
+                    viewModel.model.color.opacity(0.5).edgesIgnoringSafeArea(.all)
```
Comment (eldesouky) [inline Shared/Views/NewSubView.swift]: Remove opacity. We should display chosen color as is.

**Review by eldesouky — CHANGES_REQUESTED** [review]

File: Shared/Views/NewSubView.swift (hunk 1 above)
Comment (ahmed93) [reply Shared/Views/NewSubView.swift:152]: we can do it like u said if we won't use sheet .. but for now I will leave it as it is till we decide how to view it ..

File: Shared/Views/NewSubView.swift (hunk 4 above)
Comment (ahmed93) [reply Shared/Views/NewSubView.swift]: ya we could have done it with protocol, check the current handlation ..

**Review by eldesouky — APPROVED** [review]

**Pull request merged**